coax/td_learning/_clippeddoubleqlearning.py

```diff
@@ -139,143 +138,101 @@ def __init__(
         elif len(self.q_targ_list) * len(self.pi_targ_list) < 2:
             raise ValueError("len(q_targ_list) * len(pi_targ_list) must be at least 2")
 
-        def loss_func(params, target_params, state, target_state, rng, transition_batch):
+        # persist DoubleQLearning jit functions
+        self._super_grads_and_metrics_func = self._grads_and_metrics_func
+        self._super_td_error_func = self._td_error_func
+
+        def q_targ_pi_argmin(target_params, target_state, rng, transition_batch):
             rngs = hk.PRNGSequence(rng)
-            S = self.q.observation_preprocessor(next(rngs), transition_batch.S)
-            A = self.q.action_preprocessor(next(rngs), transition_batch.A)
-            W = jnp.clip(transition_batch.W, 0.1, 10.)  # clip importance weights to reduce variance
 
-            # regularization term
-            if self.policy_regularizer is None:
-                regularizer = 0.
+            def q_targ_target_func(p, s):
+                return self.q_targ_func(p, s, rng=next(rngs), transition_batch=transition_batch)
+            targets = jax.vmap(q_targ_target_func)(*stack_trees(target_params, target_state))
```

Comment:
Actually, I don't think this is right. You need to vmap twice, once over `q_targ_list` and then separately also over `pi_targ_list`. The vmapped output should have shape: `(len(pi_targ_list), len(q_targ_list), ...)`.

See also the toy example I posted earlier.

Reply:
One more thing, ClippedDoubleQLearning does something slightly different from DoubleQLearning in that it evaluates on the `q_targ`'s, whereas DoubleQLearning evaluates in `q` instead.

Reply:
I was wondering whether this was the way to go. Because it solved my problem but I had some trouble in understanding why it did that. I will fix this! Thank you for reviewing my code!

Reply:
I saw this difference but was not sure which way was the correct one. Many implementation just use two target networks and take the min over them. I know that evaluating on the `q_targ`s will reduce the maximization bias even further. And this makes the `q`/`q_targ` "twin" pairs more explicit. So I will change it to evaluating on `q_targ`.